core/tests/protractor_utils/LibraryPage.js

```diff
@@ -150,14 +150,42 @@ var LibraryPage = function(){
 
   this.findExploration = function(explorationTitle) {
     waitFor.pageToFullyLoad();
-    searchInput.clear();
-    searchInput.sendKeys(explorationTitle);
+    // The library page has two search bar input elements.
+    // The first input element is visible only in a desktop
+    // browser and is invisible in case of a mobile browser.
+    // The second input element is visible when the library
+    // page is rendered for mobile device.
+
+    var desktopSearchInput = searchInputs.first();
```

Comment:
Might as well define these vars in the relative branches. Split the doc above and put it in each branch:

`In mobile devices, the search input ...`

`In desktop devices, the search input is represented in the ...`

(say)

Reply:
Done, thanks!